google/resource_compute_region_autoscaler.go

```diff
@@ -0,0 +1,372 @@
+		if d.Get(prefix+"cpu_utilization.0.target").(float64) != 0 {
+			cpuUtilCount := d.Get(prefix + "cpu_utilization.#").(int)
+			if cpuUtilCount != 1 {
+				return nil, fmt.Errorf("The autoscaling_policy must have exactly one cpu_utilization, found %d.", cpuUtilCount)
+			}
+			policyCounter++
+			scaler.AutoscalingPolicy.CpuUtilization = &compute.AutoscalingPolicyCpuUtilization{
+				UtilizationTarget: d.Get(prefix + "cpu_utilization.0.target").(float64),
+			}
+		}
+	}
+	if _, ok := d.GetOk("autoscaling_policy.0.metric"); ok {
+		if d.Get(prefix+"metric.0.name") != "" {
+			policyCounter++
+			metricCount := d.Get(prefix + "metric.#").(int)
+			if metricCount != 1 {
+				return nil, fmt.Errorf("The autoscaling_policy must have exactly one metric, found %d.", metricCount)
+			}
+			scaler.AutoscalingPolicy.CustomMetricUtilizations = []*compute.AutoscalingPolicyCustomMetricUtilization{
+				{
+					Metric:                d.Get(prefix + "metric.0.name").(string),
+					UtilizationTarget:     d.Get(prefix + "metric.0.target").(float64),
+					UtilizationTargetType: d.Get(prefix + "metric.0.type").(string),
+				},
+			}
+		}
+
+	}
+	if _, ok := d.GetOk("autoscaling_policy.0.load_balancing_utilization"); ok {
+		if d.Get(prefix+"load_balancing_utilization.0.target").(float64) != 0 {
+			policyCounter++
+			lbuCount := d.Get(prefix + "load_balancing_utilization.#").(int)
+			if lbuCount != 1 {
+				return nil, fmt.Errorf("The autoscaling_policy must have exactly one load_balancing_utilization, found %d.", lbuCount)
+			}
+			scaler.AutoscalingPolicy.LoadBalancingUtilization = &compute.AutoscalingPolicyLoadBalancingUtilization{
+				UtilizationTarget: d.Get(prefix + "load_balancing_utilization.0.target").(float64),
+			}
+		}
+	}
+
+	if policyCounter != 1 {
+		return nil, fmt.Errorf("One policy must be defined for an autoscaler.")
+	}
+
+	return scaler, nil
+}
+
+func resourceComputeRegionAutoscalerCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	project, err := getProject(d, config)
+	if err != nil {
+		return err
+	}
+
+	// Get the region
+	log.Printf("[DEBUG] Loading region: %s", d.Get("region").(string))
+	region, err := config.clientCompute.Regions.Get(
+		project, d.Get("region").(string)).Do()
+	if err != nil {
+		return fmt.Errorf(
+			"Error loading region '%s': %s", d.Get("region").(string), err)
+	}
+
+	scaler, err := buildAutoscaler(d)
+	if err != nil {
+		return err
+	}
+
+	op, err := config.clientCompute.RegionAutoscalers.Insert(
+		project, region.Name, scaler).Do()
+	if err != nil {
+		return fmt.Errorf("Error creating Autoscaler: %s", err)
+	}
+
+	// It probably maybe worked, so store the ID now
+	d.SetId(scaler.Name)
+
+	err = computeOperationWait(config, op, project, "Creating Autoscaler")
+	if err != nil {
+		return err
+	}
+
+	return resourceComputeRegionAutoscalerRead(d, meta)
+}
+
+func flattenRegionAutoscalingPolicy(policy *compute.AutoscalingPolicy) []map[string]interface{} {
```

Comment:
I'm not sure this needs to be redefined for the regional autoscaler since they both use compute.Autoscaler... anyway. I didn't touch this function actually, just renamed it.

Reply:
You can delete this method and reuse the one in the non-regional version